src/redux/data/dataActions.js

```diff
@@ -0,0 +1,33 @@
+import { FETCH_DATA_FAILURE, FETCH_DATA_SUCCESS } from './dataTypes'
+
+export const fetchDataSuccess = (data) => {
+  return {
+    type: FETCH_DATA_SUCCESS,
+    payload: data,
+  }
+}
+
+export const fetchDataFailure = () => {
+  return {
+    type: FETCH_DATA_FAILURE,
+  }
+}
+
+const buildUrl = (description, location) => {
+  let url =
+    'https://cors-anywhere.herokuapp.com/https://jobs.github.com/positions.json?'
+  url += description ? `&description=${description}` : ''
```

Comment:
potresti scriverla in modo piu elegante usando array
```typescript
let url = 'https://cors-anywhere.herokuapp.com/https://jobs.github.com/positions.json'

const params = Object.entries({ description: 'desc', location: '', test: 'asd' })
  .map(([k, v]) => v ? `${k}=${encodeURI(v)}` : '')
  .filter(v => !!v)
  .join('&')

return [url, params].join('?')
```

Reply:
avrebbe anche senso muovere questa funzione da actions a un file `shared.js` o una folder `shared/buildUrl.js`